common/djangoapps/third_party_auth/signals/tests/test_handlers.py

```diff
@@ -33,100 +33,77 @@ def setUp(self):
-        # Verifies that basic attributes are set but no provider updates are attempted
-        {
-            'enabled': False,
-            'simulate_error': False,
-            'description': 'handlers disabled',
-            'expected_calls': [
+    @mock.patch('common.djangoapps.third_party_auth.signals.handlers.set_custom_attribute')
+    def test_saml_config_signal_handlers_disabled(self, mock_set_custom_attribute):
+        """
+        Test behavior when SAML config signal handlers are disabled.
+
+        Verifies that basic attributes are set but no provider updates are attempted.
+        """
+        with override_settings(ENABLE_SAML_CONFIG_SIGNAL_HANDLERS=False):
+            self.saml_config.entity_id = 'https://updated.example.com'
+            self.saml_config.save()
+
+            expected_calls = [
                 call('saml_config_signal.enabled', False),
-                call('saml_config_signal.new_config_id', 'CONFIG_ID'),
+                call('saml_config_signal.new_config_id', self.saml_config.id),
                 call('saml_config_signal.slug', 'test-config'),
-            ],
-            'expected_call_count': 3,
-        },
-        # Case 2: Tests behavior when SAML config signal handlers are enabled
-        # Verifies that attributes are set and provider updates are attempted successfully
-        {
-            'enabled': True,
-            'simulate_error': False,
-            'description': 'handlers enabled',
-            'expected_calls': [
-                call('saml_config_signal.enabled', True),
-                call('saml_config_signal.new_config_id', 'CONFIG_ID'),
-                call('saml_config_signal.slug', 'test-config'),
-                call('saml_config_signal.updated_count', 0),
-            ],
-            'expected_call_count': 4,
-        },
-        # Case 3: Tests error handling when signal handlers are enabled but encounter an exception
-        # Verifies that error information is properly captured when provider updates fail
-        {
-            'enabled': True,
-            'simulate_error': True,
-            'description': 'handlers enabled with exception',
-            'expected_calls': [
-                call('saml_config_signal.enabled', True),
-                call('saml_config_signal.new_config_id', 'CONFIG_ID'),
-                call('saml_config_signal.slug', 'test-config'),
-            ],
-            'expected_call_count': 4,  # includes error_message call
-            'error_message': 'Test error',
-        },
-    )
-    @ddt.unpack
+            ]
+
+            # Verify expected calls were made
+            mock_set_custom_attribute.assert_has_calls(expected_calls, any_order=False)
+
+            # Verify total call count
+            assert mock_set_custom_attribute.call_count == 3, (
+                f"Expected 3 calls for disabled handlers, "
+                f"got {mock_set_custom_attribute.call_count}"
+            )
```

Comment:
@ktyagiapphelix2u: I'm making this more compact.
```suggestion
            mock_set_custom_attribute.assert_has_calls(expected_calls, any_order=False)
            assert mock_set_custom_attribute.call_count == 3
```